src/asmjs/asm_v_wasm.cpp

```diff
@@ -27,21 +27,25 @@ Type asmToWasmType(AsmType asmType) {
     case ASM_FLOAT: return Type::f32;
     case ASM_INT64: return Type::i64;
     case ASM_NONE: return Type::none;
-    default: {}
+    case ASM_FLOAT32X4:
+    case ASM_FLOAT64X2:
+    case ASM_INT8X16:
+    case ASM_INT16X8:
+    case ASM_INT32X4: WASM_UNREACHABLE();
   }
-  abort();
+  WASM_UNREACHABLE();
```

Comment:
Are these two types of unreachability really the same?   The latter means something like data corruption happened to break the type system and given the enum a bad value, whereas the former is maybe bad input?  (or maybe not?)

Reply:
I agree that WASM_UNREACHABLE is not the right way to deal with bad input. I was careful in the binary parser to keep defaults for when we read junk bytes. In this particular instance, it seemed safe enough to assume that this function is being passed a valid `AsmType`.